src/benchmarklib/benchmark_utils.cpp

```diff
@@ -31,28 +31,41 @@ std::ostream& get_out_stream(const bool verbose) {
 }
 
 BenchmarkState::BenchmarkState(const size_t max_num_iterations, const opossum::Duration max_duration)
-    : max_num_iterations(max_num_iterations), max_duration(max_duration) {}
+    : max_num_iterations(max_num_iterations), max_duration(max_duration) {
+  iteration_durations.reserve(max_num_iterations);
+}
 
 bool BenchmarkState::keep_running() {
+  bool is_first_iteration = false;
+
   switch (state) {
     case State::NotStarted:
-      begin = std::chrono::high_resolution_clock::now();
+      benchmark_begin = std::chrono::high_resolution_clock::now();
       state = State::Running;
+      is_first_iteration = true;
       break;
     case State::Over:
       return false;
     default: {}
   }
 
+  const auto now = std::chrono::high_resolution_clock::now();
+  const auto benchmark_duration = now - benchmark_begin;
+  const auto iteration_duration = now - iteration_begin;
+
+  if (!is_first_iteration) {
+    iteration_durations.push_back(iteration_duration);
+  }
+
+  benchmark_end = now;
+  iteration_begin = now;
```

Comment:
> We might want to have a second section in our DEPENDENCIES.md for python dependencies.

I don't see how that list would help anyone. If anything, we should write a script like
`pip install beautifultable  # for ...`

Reply:
Any reason against a `requirements.txt` and `pip install -r requirements.txt`?
That's standard approach for Python dependencies afaik.

Reply:
I added `requirements.txt` containing
```
scipy>=1.0
termcolor>=1.0
terminaltables>=3.0
```
Anything else?
And should we keep it as `requirements.txt` or change the name?

Reply:
Since the name is well accepted and de-facto the standard name for the requirements file, I would vote to keep the name.